### PR TITLE
add pymesh2

### PR DESCRIPTION
**Pull request by ocefpaf**
@conda-forge/staged-recipes this recipe is a "study case" for the mess in the packaging ecosystem and do bother with a serious review here. (We need this for... reasons, and a source build will follow up soon.)

Ping @meredithmclinn who will be able to install this with `conda` on Linux+Python 3.6 once we merge it.

### CODE REVIEW

File: recipes/pymesh2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.2.1" %}
+
+package:
+  name: pymesh2
+  version: {{ version }}
+
+source:
+  url: https://github.com/PyMesh/PyMesh/releases/download/v{{ version }}/pymesh2-{{ version }}-cp36-cp36m-linux_x86_64.whl
```
Comment (isuruf) [inline recipes/pymesh2/meta.yaml]: This is not even a `manylinux1` wheel. Why not use `pip` to install this if there's an urgent need?

Comment (ocefpaf) [reply recipes/pymesh2/meta.yaml]: Reasons... (I'll ping me in pvt and I'll explain it.)

File: recipes/pymesh2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - python
+    - pip
+  run:
+    - python
+    - numpy
+    - scipy
+
+test:
+  imports:
+    - pymesh  # [not linux]
```
Comment (isuruf) [inline recipes/pymesh2/meta.yaml]: skipped for linux?

Comment (ocefpaf) [reply recipes/pymesh2/meta.yaml]: Yep. Just to fool the linter b/c it is GLIBC 2.14 and we don't have that. (The tests passes in a local installation of this artifact though.)

This is a study case b/c this package touches virtually everything we discuss here during the Package Summit, so fixing ti later is a way to show how we do things in conda-forge.

File: recipes/pymesh2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+about:
+  home: https://github.com/qnzhou/PyMesh
```
Comment (isuruf) [inline recipes/pymesh2/meta.yaml]: Wrong homepage

File: recipes/pymesh2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  license: MPL
+  # license_file:
```
Comment (isuruf) [inline recipes/pymesh2/meta.yaml]: Need a license file

File: recipes/pymesh2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    - numpy
+    - scipy
+
+test:
+  imports:
+    - pymesh  # [not linux]
+
+about:
+  home: https://github.com/qnzhou/PyMesh
+  license: MPL
```
Comment (isuruf) [inline recipes/pymesh2/meta.yaml]: This package is not MPL even though the authors claim that it is. It's shipping GPL, LGPL, Apache2, and non open-source (triangle) components.

Comment (ocefpaf) [reply recipes/pymesh2/meta.yaml]: I opened an issue upstream.

**Pull request merged**